### PR TITLE
Flag documentation

### PR DESCRIPTION
**Pull request by 12AwsomeMan34**
Documentation on flags and their usage.

Point out if anything needs to be clarified or expanded upon.

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/cb83534a/index.html.

Here are some links to the pages that were modified:

- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/cb83534a/plugin/commands/arguments.html
- modified https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/cb83534a/plugin/commands/index.html
- added https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/cb83534a/plugin/commands/flags.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/474#issuecomment-203672834), not to the direct url to the preview._

File: source/plugin/commands/flags.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    if (optional.isPresent()) {
+        int value = optional.get().intValue();
+    } else {
+        src.sendMessage(Text.of("The value flag was not specified."));
+    }
+
+Long Flags
+==========
+
+Alternatively to short flags like the ones we have been using above, we can also use long flags. Using a long flag,
```
Comment (Inscrutable) [inline source/plugin/commands/flags.rst]: Icky sentence. Try "As an alternative to" ...

File: source/plugin/commands/flags.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+To create a flag, we first need a builder for flags. We can simply use the ``flags()`` method provided by
+``GenericArguments`` to obtain the builder we need. From there, we can specify what type of flag we would like to
+create. Note that flags are specified as an argument.
+
+.. code-block:: java
+    
+    import org.spongepowered.api.command.args.GenericArguments;
+    
+    .arguments(GenericArguments.flags().flag("s").buildWith(GenericArguments.none()))
```
Comment (RobertHerhold) [inline source/plugin/commands/flags.rst]: I would rather have the full compiling example, ie `CommandSpec.builder().arguments()...`

File: source/plugin/commands/flags.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+This will create a command flag, so that when the player performs ``/our-command -s``, the flag for ``s`` will be true.
+Note the ``GenericArguments.none()``. This will prevent the command from having any arguments. If you wish for the
```
Comment (RobertHerhold) [inline source/plugin/commands/flags.rst]: Combine these sentences please - "Note that `GenericArguments.none()` will prevent..."

**Comment by RobertHerhold** [comment]
@12AwsomeMan34 Looks good apart from mine and Inscrutable's comments. Feel free to merge after you revise.

File: source/plugin/commands/flags.rst (hunk 2 above)
Comment (12AwsomeMan34) [inline source/plugin/commands/flags.rst]: Just this one example, or all of them? Or maybe have an 'in conclusion' at the bottom with everything from above?

Comment (RobertHerhold) [inline source/plugin/commands/flags.rst]: I think all of them would be appropriate

**Pull request merged**